### ISSUE TITLE
Could we get an updated list of LZ joseki?

### ISSUE THREAD

**Issue by anonymousAwesome** [gcp/leela-zero#2055]
http://zero.sjeng.org/opening hasn't been updated in 4 months.

**Comment by jillybob** [comment]
I'd like this too.

**Comment by roy7** [comment]
@rchs819 if you are still messing with LZ, could you maybe PR another joseki/opening update? :)

**Comment by sethtroisi** [comment]
@mentioned user hasn't had any updates on github since august
@roy7 Are you interested in owning this or should I close it?

**Comment by roy7** [comment]
Hmm I'm not sure how to actually generate this is the issue. I think @rchs819 always did them himself and then did a PR to update the server. We'd need someone to figure out what commands to run to generate new data. :) Perhaps long term it could be automated to run nightly or something.

**Comment by amj** [comment]
this is a cool feature... it'd be awesome to be able to bring it back!

**Comment by alreadydone** [comment]
https://www.bestopenbook.com/archives/811 #153 net, 200k visits for most visited move
https://www.bestopenbook.com/ mostly using LZ to analyze pro game / AI game openings

**Comment by sethtroisi** [comment]
I'd love to run whatever tool they was using to create the analysis on minigo SGFs.
@roy7 It looks like @rchs819 didn't leave a branch for creating the openings, but they do have a reddit post about it https://www.reddit.com/r/cbaduk/comments/8lxrgp/leela_zero_opening_study_things_got_weird_lately/

I'll try running this over the weekend

**Comment by roy7** [comment]
Yeah I forgot it was a tool merged into the repo https://github.com/leela-zero/leela-zero-server/pull/120. I've never ran it myself. In a perfect world we'd automate it. :)

**Comment by sethtroisi** [comment]
@roy7 in this code do you know what obj.B.hash is?
```
const cursor = db.collection("opening").find();
while ((obj = await cursor.nextObject())) {
...
    const b_training = hashes[obj.B.hash] || 0;
```

**Comment by roy7** [comment]
I'm afraid this bit of low level detail I'm clueless about. Maybe someone like @killerducky or @Mardak can come to the rescue. @marcocalignano also knows a lot about how the SGF files get made.

**Comment by alreadydone** [comment]
Not sure what you are asking, but `slice(0,6)` seems to correspond to the 6-digit network hash at http://zero.sjeng.org/opening

**Comment by sethtroisi** [comment]
I think I understand the code now.

in parse_players, set
game["B"] = player_name.split(" ")[1][:6]
game["W"] = player_name.split(" ")[1]:6]

At the start of [opening_analysis](https://github.com/leela-zero/leela-zero-server/blob/next/scripts/opening_analysis.js#L163) 
look up all the network hashes and save them in 
hashes[\<six digits\>] = number_of_games

final for each sgf
lookup each player name (my confusion is why it's looking up PW here)
b_training (which is really b_training_steps) = hashes[PW]

this allows us to save the training step (the Y axis) for this sgf.

**Comment by roy7** [comment]
If you end up getting this working well @sethtroisi maybe you could run a pass on LZ games too and PR the update to the server's openings page? ;)

**Comment by sethtroisi** [comment]
That's what I'm trying :)
